### PR TITLE
The application structure and functionality

### PR DESCRIPTION
**Pull request by hatik92**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_APP_NAME="React Quantum"
+REACT_APP_LEARN_MORE="https://quantum.softberg.org"
+REACT_APP_API_URL="backEnd url"
```
Comment (armanist) [inline .env.example:3]: What this constant is for : REACT_APP_API_URL ?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
```
Comment (armanist) [inline .gitignore:2]: Ignore the package-lock.json

File: public/custom.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+jQuery(document).ready(function ($) {
```
Comment (armanist) [inline public/custom.js]: Why do we need jQuery do be used??

File: public/index.html [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	<meta name="theme-color" content="#000000" />
+	<meta name="description" content="Web site created using create-react-app" />
+	<link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+	<!--
+      manifest.json provides metadata used when your web app is installed on a
+      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
+    -->
+	<link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+	<!--
+      Notice the use of %PUBLIC_URL% in the tags above.
```
Comment (armanist) [inline public/index.html]: Remove all unnecessary comments

File: public/manifest.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (armanist) [inline public/manifest.json]: What this file is doing?

File: public/robots.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
```
Comment (armanist) [inline public/robots.txt:1]: What this file is doing?

File: src/App.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import './App.css';
+import { useEffect } from 'react';
+import { Routes, Route } from "react-router-dom";
+import Home from './components/Home/Home';
+import NotFound from './NotFound/NotFound';
+import Posts from './components/Posts/Posts';
```
Comment (armanist) [inline src/App.js]: Why it's needed to create folders for each single file components?

File: src/App.test.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
```
Comment (armanist) [inline src/App.test.js]: Does this test working?

File: src/api/api.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+	response => response,
+	error => {
+		if (!error.response) {
+			console.log('?', error);
+		}
+		if (error.response.status === 401) {
+			localStorage.clear()
+		}
+		// if (error.response.status === 404) {
+		// 	return
```
Comment (armanist) [inline src/api/api.js]: Remove unnecessary comments

File: src/api/api.js [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+				console.log(err);
+			}
+			break;
+
+		case 'POST':
+			try {
+				if (tokens.access_token !== null) {
+					return await apiConfig.post(url, postDataOrGetParam, { headers: headers(tokens) })
+				} else if (routesWithoutToken.includes(url)) {
+					return await apiConfig.post(url + token, postDataOrGetParam)
```
Comment (armanist) [inline src/api/api.js]: Why we pass token, if the route is without token?

File: src/api/api.js [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+				console.log(err);
+			}
+			break;
+		case 'PUT':
+
+			try {
+				if (tokens.access_token !== null) {
+					return await apiConfig.put(url, postDataOrGetParam, { headers: headers(tokens) })
+				} else if (routesWithoutToken.includes(url)) {
+					return await apiConfig.post(url, postDataOrGetParam)
```
Comment (armanist) [inline src/api/api.js]: Why we do post if we are in put block?

File: src/components/Home/Home.jsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+						</div>
+					</div>
+					<div className="index-links">
+						<Link to="/about" className="white-text">About</Link>
+						<a href={process.env.REACT_APP_LEARN_MORE} target="_blank" rel="noreferrer" className="white-text">Learn More</a>
+					</div>
+				</div>
+			</div>
+		</div>
+		<ul className="bg-bubbles">
```
Comment (armanist) [inline src/components/Home/Home.jsx]: Bubbles can be moved to separate component

File: src/components/About/About.jsx [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+								<li>
+									<h6>Run the project</h6>
+									<code>&gt; php qt serve</code>
+								</li>
+							</ul>
+						</div>
+					</div>
+				</div>
+			</div>
+			<ul className="bg-bubbles">
```
Comment (armanist) [inline src/components/About/About.jsx]: Bubbles can me moved to separate component

File: src/components/Auth/RequireAuth.js [hunk 14]
```diff
@@ -0,0 +1,13 @@
+import { useLocation, Navigate, Outlet } from 'react-router-dom';
+
+const RequireAuth = () => {
+	const location = useLocation()
+
+	return (
+		localStorage.getItem("isAuth")
```
Comment (armanist) [inline src/components/Auth/RequireAuth.js]: Make sure the isAuth really needed

File: src/components/Auth/SignIn.jsx [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+					setTokens(res.data.tokens)
+					authMe(setAuth, navigate, returnLocation)
+				}
+				if (res.data.code) {
+					setAuth({ code: res.data.code })
+					navigate('/verify', { replace: true, state: { from: { pathname: location.state?.from?.pathname } } });
+				}
+			} else if (res.data.status === 'error') {
+				switch (res.data.message) {
+					case "inactive_account":
```
Comment (armanist) [inline src/components/Auth/SignIn.jsx]: No need to check the message, it's back-end issue

File: src/reportWebVitals.js [hunk 16]
```diff
@@ -0,0 +1,13 @@
+const reportWebVitals = onPerfEntry => {
```
Comment (armanist) [inline src/reportWebVitals.js]: What this file doing?

File: src/setupTests.js [hunk 17]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
```
Comment (armanist) [inline src/setupTests.js]: Is this file needed?

**Review by armanist — CHANGES_REQUESTED** [review]

File: public/manifest.json (hunk 5 above)
Comment (hatik92) [reply public/manifest.json]: The manifest.json provides information about a web application in a JSON text file, necessary for the web app to be downloaded and be presented to the user similarly to a native app (e.g., be installed on the homescreen of a device, providing users with quicker access and a richer experience). PWA manifests include its name, author, icon(s), version, description, and list of all the necessary resources (among other things).

**Review by armanist — APPROVED** [review]

**Review by armanist — APPROVED** [review]

**Pull request merged**